datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -478,7 +481,7 @@ func (f *ChainDataFetcher) retryFunc(insert HandleChainEventFn) HandleChainEvent
 		for err := insert(event, reqType); err != nil; err = insert(event, reqType) {
 			select {
 			case <-f.stopCh:
-				return nil
+				return err
```

Comment:
Why do you return `err`? It is the return value of `insert()`.
```
var dataNotFoundErr = errors.New("chaindatafetcher is stopped by user")
```
``` suggestion
				return dataNotFoundErr
```

Reply:
@winnie-byun The only failure case is when the user stops chaindatafetcher while retrying. The outer function leaves log with the error message.

Reply:
Thanks for the explanation :)